pkg/kratos/handlers.go

```diff
@@ -79,6 +80,18 @@ func (a *API) handleCreateFlow(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	flow, err = a.service.FilterFlowProviderList(context.Background(), flow)
+	if err != nil {
+		a.logger.Errorf("Error when filtering providers: %v\n", err)
+		switch err {
+		// If no providers are allowed, we simply log the error and return an empty flow
+		case ErrNoProvidersAllowed:
+		default:
+			http.Error(w, "Unexpected error", http.StatusInternalServerError)
+			return
+		}
+	}
+
```

Comment:
this seems a bit odd, i know but cant make it any better as struggling to get my head around it but https://github.com/canonical/identity-platform-login-ui/pull/208/files#diff-5e4ba31969008b305d67765d5d4c9bd50d9bcc49571f1594e97d2b152ddf1378R27 doesn't look right and this code flow goes around that concept....not sure

Reply:
Since we don't treat it as an error, it probably should not be an error. It just feels weird from a UX perspective not returning any providers, so I added the error there in case we need to add some special logic in the future.

I will remove it